src/components/Docs/SnippetViewer/BatchCheckRequestViewer.tsx

```diff
@@ -108,62 +108,71 @@ const { result } = await fgaClient.batchCheck(body, options);
 */`;
 
     case SupportedLanguage.GO_SDK:
-      return `// The Go SDK does not yet support server-side batch checks. This currently just calls the check endpoint in parallel.
+      return `// Requires >=v0.7.0 for the server side BatchCheck, earlier versions support a client-side BatchCheck with a slightly different interface
+body := ClientBatchCheckRequest{
+  Checks: []ClientBatchCheckItem{${checks
+    .map(
+      (check) => `
+    {
+      User:          "${check.user}",
+      Relation:      "${check.relation}",
+      Object:        "${check.object}",
+      CorrelationId: "${check.correlation_id}",${
+        check.contextualTuples
+          ? `
+      ContextualTuples: []ClientContextualTupleKey{
+        ${check.contextualTuples
+          .map(
+            (tuple) => `{
+          User:     "${tuple.user}",
+          Relation: "${tuple.relation}",
+          Object:   "${tuple.object}",
+        },`,
+          )
+          .join('\n        ')}
+      },`
+          : ''
+      }${
+        check.context
+          ? `
+      Context: &map[string]interface{}${JSON.stringify(check.context)},`
+          : ''
+      }
+    },`,
+    )
+    .join('')}
+  },
+}
+
+options := BatchCheckOptions{
+  MaxBatchSize:         PtrInt32(50), // optional, default is 50, can be used to limit the number of checks in a single server request
+  MaxParallelRequests:  PtrInt32(10), // optional, default is 10, can be used to limit the parallelization of the BatchCheck chunks${
+    modelId
+      ? `,
+  AuthorizationModelId: PtrString("${modelId}")`
+      : ''
```

Comment:
I think these should be using `openfga.PtrInt32` similar to the check code as otherwise I get errors on conflicting imports. Also needs a trailing comma on the model id

```suggestion
  MaxBatchSize:         openfga.PtrInt32(50), // optional, default is 50, can be used to limit the number of checks in a single server request
  MaxParallelRequests:  openfga.PtrInt32(10), // optional, default is 10, can be used to limit the parallelization of the BatchCheck chunks${
    modelId
      ? `,
  AuthorizationModelId: openfga.PtrString("${modelId}")`,
      : ''
```

Reply:
Applied, Thanks @ewanharris